### DISCUSSION TITLE
Pockethost 0.3.0 released: new UX, dark mode, realtime support, docker, and more

### DISCUSSION THREAD

**Discussion by benallfree** [pocketbase/pocketbase#842] [Show and tell]
[PocketHost.io](https://pockethost.io), the zero-config hosting platform for PocketBase, has reached v0.3.0.

New in this release:
* Completely overhauled UX (big thank you @brewhousedigital for this contribution)
* Completely refactored Docker strategy for more defined dev and prod experience
* Realtime improvements (thank you @rajesh6161 for the report and testing)

Some stats:
* 64 users
* 75 instances created

I might have more interesting stats in the next round. Right now it's difficult to tell how many instances are active over what period. We have an issue open to tackle that https://github.com/benallfree/pockethost/issues/21.

I really appreciate everyone's support and feedback. Adding support for TS/JS cloud functions remains at the top of my list. I need it for a personal project game I'm developing. I currently have backend providers written in Firebase and Supabase. PocketBase is juuuust a little behind in a couple key areas needed for realtime presence detection and server-side events needed to preserve data integrity when users disconnect. But if I can get TS/JS cloud functions and an additional realtime disconnect hook, then I'll be able to move off Firebase altogether. 

As always, every modification I make to the PocketBase core is submitted as a PR in case it is something that @ganigeorgiev wants to incorporate. That said, I admire and appreciate his focus on keeping the core uncluttered. Many of the enhancements I'm making will likely live outside PocketBase. It's beginning to look like the pockethost monorepo will be home to such enhancements.

pockethost.io was inspired by loving PocketBase but still wanting a managed backend experience. Although managed hosting might at first seem antethetical to the PocketBase core philosophy, I think there is a BaaS experience that can align with the goal of sovereignty at the same time. That's my focus with pockethost. I suspect there are many developers like me who would like the approach provided by PocketBase but who really want the BaaS experience of Firebase/Supabase (and additionally the promise of portability!).

PocketHost is a 100% open source project. You can contribute or modify it to suit at https://github.com/benallfree/pockethost.

Thank you for the support and thank you @ganigeorgiev for the great core project and for letting me post occasionally about what I'm building with PocketBase. It's almost Novemeber, and I'm very excited about your transition to full-time focus on PB.

**Comment by rajesh6161** [comment]
@benallfree The website now looks freakingly beautiful and organized. Keep up the good work 🎉 and keep going 🚀.

**Reply by benallfree** [reply]
All credit to @brewhousedigital for the new design!

**Comment by kopyl** [comment]
Dude, you won't believe, I just came up with the idea of creating a cloud PocketBase SaaS.

You're the best :)

We might chat in Telegram if you would like to: @kopyl

**Comment by kopyl** [comment]
By the way, how to shut down an instance?

I just created one to check things out, but don't need it now. My email is spammmmm1997@gmail.com .

And another question – how do you handle API authorization? I mean anybody could make http request and dump your db or so. How do you avoid it?

**Reply by benallfree** [reply]
> how to shut down an instance

Idle instances are shut down automatically :)

> how do you handle API authorization? I mean anybody could make http request and dump your db or so. 

No. Try it :) By default, PocketBase allows only admin accounts to read and write data. @ganigeorgiev can you confirm though?

**Reply by ganigeorgiev** [reply]
@benallfree, @kopyl Yes, collections by default can be accessed only by admins. This can be changed by updating the individual "API rules" for each action (list/search, view, create, update, delete). More info could be found in https://pocketbase.io/docs/manage-collections/#collection-api-rules.

**Reply by kopyl** [reply]
> > how to shut down an instance
> 
> 
> 
> Idle instances are shut down automatically :)
> 
> 
> 
> > how do you handle API authorization? I mean anybody could make http request and dump your db or so. 
> 
> 
> 
> No. Try it :) By default, PocketBase allows only admin accounts to read and write data. @ganigeorgiev can you confirm though?

But when you create a connection, how do you specify credentials?)

When you instantiate it with new PocketBase(url).
The url is like https://able-intern.pockethost.io
But there is no credentials in it like in MongoDB connection URL.

**Reply by ganigeorgiev** [reply]
@kopyl Initializing the SDK (aka. `new PocketBase(url)`) doesn't perform any requests. It just creates and initialize a local JS object. When sending requests to fetch your data, the collection API rules will check server-side whether the request client has permissions to perform the action or not.

To authenticate as admin with the SDK you can call `pb.admins.authWithPassword(...)`. Similarly, to authenticate as app user, you can call `pb.collection('users').authWithPassword(...)`. For more information on the authentication, you could check https://pocketbase.io/docs/authentication/.

**Reply by kopyl** [reply]
> @kopyl Initializing the SDK (aka. `new PocketBase(url)`) doesn't perform any requests. It just creates and initialize a local JS object. When sending requests to fetch your data, the collection API rules will check server-side whether the request client has permissions to perform the action or not.
> 
> 
> 
> To authenticate as admin with the SDK you can call `pb.admins.authWithPassword(...)`. Similarly, to authenticate as app user, you can call `pb.collection('users').authWithPassword(...)`. For more information on the authentication, you could check https://pocketbase.io/docs/authentication/.

Do I need to pass a user's password or a password to a db?

**Reply by ganigeorgiev** [reply]
@kopyl The database doesn't have a password (it is just a sqlite db file). As mentioned previously, depending on your collections configuration you can implement different type of access controls to your data. By default, all new collections are with "Only admins" permissions, but you can change that from the related collection settings > API Rules in the PocketBase Admin UI (you can even allow full guest/public access).

To authenticate you need to send `Authorization: TOKEN` header with each request.

The `TOKEN` could be admin or app user token. When using the SDKs, as shown in https://pocketbase.io/docs/authentication/, the `Authorization` header is automatically added for you once authenticated.

If you want to generate the tokens manually via the HTTP API:
- for admin token, check the "Auth with password" section in https://pocketbase.io/docs/api-admins/
- for app user token, check the "Auth with password/Auth with OAuth2" sections in https://pocketbase.io/docs/api-records/#auth-record-actions

**Reply by kopyl** [reply]
Thanks :)
